File: src/main/java/no/fint/MailingService.java
package no.fint;

import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

import javax.mail.MessagingException;
import javax.mail.Transport;
import javax.mail.internet.MimeMessage;
import java.util.Date;

@Service
@Slf4j
public class MailingService {

    private final Config config;

    public MailingService(Config config) {
        this.config = config;
    }

    public boolean send(String content) {
        try {
            log.info("Creating email from {} to {} ...", config.getSender(), config.getRecipients());
            Transport.send(createMimeMessage(content));
            return true;
        } catch (MessagingException e) {
            logSendEmailError(e);
            return false;
        }
    }

    private MimeMessage createMimeMessage(String content) throws MessagingException {
        String subject = generateSubject();
        return new FintMimeMessage(config, content, subject).getMail();
    }

    private String generateSubject() {
        return String.format("Org Monitor %TF %<TR", new Date());
    }

    private void logSendEmailError(MessagingException e) {
        log.error("Unable to send message! smtpUsername: {} smtpHost: {} smtpPort: {}",
                config.getSmtpUsername(), config.getSmtpServer(), config.getSmtpPort(), e);
    }
}

File: src/main/kotlin/no/fint/mailing/MailingService.kt
package no.fint.mailing

import no.fint.Config
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import javax.mail.MessagingException
import javax.mail.Transport
import javax.mail.internet.MimeMessage

@Service
class MailingService(
    private val config: Config,
) {
    private val logger = LoggerFactory.getLogger(javaClass)

    fun send(content: String): Boolean {
        logger.info("Sending email from ${config.sender} to ${config.recipients}")
        try {
            Transport.send(createMimeMessage(content))
        } catch (e: MessagingException) {
            logSendEmailError(e)
            return false
        }
        return true
    }

    private fun createMimeMessage(content: String): MimeMessage = FintMimeMessage(config, content, generateSubject()).mail

    private fun generateSubject() =
        java.time.LocalDateTime.now().let { now ->
            "Org Monitor for ${config.orgid} ${now.toLocalDate()} ${now.toLocalTime().withNano(0) }"
        }

    private fun logSendEmailError(e: MessagingException) {
        logger.error(
            "Unable to send message! smtpUsername: ${config.smtpUsername}, smtpHost: ${config.smtpServer}, smtpPost: ${config.smtpPort}, Error: $e",
        )
    }
}
